File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/Java/C/s893190600_comments_removed.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

#define MAX 1024 * 8
#define MAX_STR 500 * 8 * 2

char buffer[MAX];
char str[MAX_STR];
bool isDigit[256];
bool isSpace[256];
bool isLineSep[256];
int pos = 0;
int readLen = 0;

void init() {
    for (int i = 0; i < 10; i++) {
        isDigit['0' + i] = true;
    }
    isDigit['-'] = true;
    isSpace[' '] = isSpace['\r'] = isSpace['\n'] = isSpace['\t'] = true;
    isLineSep['\r'] = isLineSep['\n'] = true;
}

int read() {
    if (pos >= readLen) {
        pos = 0;
        readLen = fread(buffer, sizeof(char), MAX, stdin);
        if (readLen <= 0) {
            exit(0);
        }
    }
    return buffer[pos++];
}

int nextInt() {
    int len = 0;
    str[len++] = read();
    len = reads(len, isSpace);
    int i = 0;
    int ret = 0;
    if (str[0] == '-') {
        i = 1;
    }
    for (; i < len; i++) ret = ret * 10 + str[i] - '0';
    if (str[0] == '-') {
        ret = -ret;
    }
    return ret;
}

long nextLong() {
    int len = 0;
    str[len++] = read();
    len = reads(len, isSpace);
    int i = 0;
    long ret = 0L;
    if (str[0] == '-') {
        i = 1;
    }
    for (; i < len; i++) ret = ret * 10 + str[i] - '0';
    if (str[0] == '-') {
        ret = -ret;
    }
    return ret;
}

double nextDouble() {
    int len = 0;
    str[len++] = read();
    len = reads(len, isSpace);
    int i = 0;
    double ret = 0;
    if (str[0] == '-') {
        i = 1;
    }
    int cnt = 0;
    for (; i < len; i++) {
        if (str[i] == '.') {
            cnt = 10;
            continue;
        }
        if (cnt == 0) {
            ret = ret * 10 + str[i] - '0';
        } else {
            ret = ret + ((double)(str[i] - '0') / cnt);
            cnt *= 10;
        }
    }
    if (str[0] == '-') {
        ret = -ret;
    }
    return ret;
}

char* nextString() {
    char* ret = (char*)malloc(MAX_STR * sizeof(char));
    strcpy(ret, nextDChar());
    ret[strcspn(ret, "\n")] = 0;
    return ret;
}

char* nextDChar() {
    int len = 0;
    len = reads(len, isSpace);
    char* ret = (char*)malloc((len + 1) * sizeof(char));
    for (int i = 0; i < len; i++) ret[i] = str[i];
    ret[len] = 0x00;
    return ret;
}

char nextChar() {
    while (true) {
        int c = read();
        if (!isSpace[c]) {
            return (char)c;
        }
    }
}

int reads(int len, bool* accept) {
    while (true) {
        int c = read();
        if (accept[c]) {
            break;
        }
        if (len == MAX_STR) {
            char* rep = (char*)malloc(MAX_STR * 3 / 2 * sizeof(char));
            memcpy(rep, str, len * sizeof(char));
            free(str);
            str = rep;
        }
        str[len++] = (char)c;
    }
    return len;
}

void prn(char* s) {
    printf("%s\n", s);
}

void prn_int(int i) {
    printf("%d\n", i);
}

void prn_long(long i) {
    printf("%ld\n", i);
}

void prr(char* s) {
    printf("%s", s);
}

int main() {
    init();
    int H = nextInt();
    int W = nextInt();
    char C[H + 2][W + 2];
    memset(C, '.', sizeof(C));
    for (int i = 0; i < H; i++) {
        char* line = nextString();
        for (int j = 0; j < W; j++) {
            C[i + 1][j + 1] = line[j];
        }
        free(line);
    }
    if (H == 1 && W == 1) {
        prn("No");
        return 0;
    }
    for (int i = 1; i <= H; i++) {
        for (int j = 1; j <= W; j++) {
            if (C[i][j] == '#' && C[i - 1][j] != '#' && C[i + 1][j] != '#' 
                && C[i][j - 1] != '#' && C[i][j + 1] != '#') {
                prn("No");
                return 0;
            }
        }
    }
    prn("Yes");
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/Java/Python/s893190600_comments_removed.py
import sys
import math
import bisect
from collections import deque, defaultdict, Counter, OrderedDict

class Solver:
    def __init__(self, input, output):
        self.input = input
        self.output = output

    def solve(self):
        H = self.ni()
        W = self.ni()
        C = [['.' for _ in range(W + 2)] for _ in range(H + 2)]
        for i in range(H):
            c = self.ns()
            for j in range(W):
                C[i + 1][j + 1] = c[j]
        if H == 1 and W == 1:
            self.prn("No")
            return
        for i in range(1, H + 1):
            for j in range(1, W + 1):
                if C[i][j] == '#' and C[i - 1][j] != '#' and C[i + 1][j] != '#' and C[i][j - 1] != '#' and C[i][j + 1] != '#':
                    self.prn("No")
                    return
        self.prn("Yes")

    def prn(self, s):
        self.output.write(f"{s}\n")

    def prr(self, s):
        self.output.write(s)

    def ni(self):
        return int(self.ns())

    def nl(self):
        return int(self.ns())

    def nd(self):
        return float(self.ns())

    def ns(self):
        return self.input.readline().strip()

    def ndi(self, n):
        return [self.ni() for _ in range(n)]

    def ndl(self, n):
        return [self.nl() for _ in range(n)]

    def ndd(self, n):
        return [self.nd() for _ in range(n)]

    def nds(self, n):
        return [self.ns() for _ in range(n)]

    def nddi(self, n, m):
        return [self.ndi(m) for _ in range(n)]

    def nddl(self, n, m):
        return [self.ndl(m) for _ in range(n)]

class PP:
    def __init__(self, key, val):
        self.key = key
        self.val = val

class PPP:
    def __init__(self, key, val1, val2):
        self.key = key
        self.val1 = val1
        self.val2 = val2

class PPL:
    def __init__(self, key, val):
        self.key = key
        self.val = val

class PPDL:
    def __init__(self, key, val):
        self.key = key
        self.val = val

    def dump(self, output):
        output.write(f"key = {self.key}  val ")
        for v in self.val:
            output.write(f"[{v}] ")
        output.write("\n")

class PPKEY:
    def __init__(self, key, val):
        self.key = key
        self.val = val

    def __eq__(self, other):
        if isinstance(other, PPKEY):
            return self.key == other.key and self.val == other.val
        return False

    def __hash__(self):
        return hash((self.key, self.val))

class PPLKEY:
    def __init__(self, key, val):
        self.key = key
        self.val = val

    def __eq__(self, other):
        if isinstance(other, PPLKEY):
            return self.key == other.key and self.val == other.val
        return False

    def __hash__(self):
        return hash((self.key, self.val))

class Pair:
    def __init__(self, key, value):
        self.key = key
        self.value = value

class MyInput:
    def __init__(self, input):
        self.input = input

    def read(self):
        return self.input.read(1)

    def nextInt(self):
        return int(self.ns())

    def nextLong(self):
        return int(self.ns())

    def nextDouble(self):
        return float(self.ns())

    def nextString(self):
        return self.ns().strip()

    def nextDChar(self):
        return list(self.ns())

    def nextChar(self):
        while True:
            c = self.read()
            if c.strip():
                return c

    def ns(self):
        while True:
            c = self.read()
            if c.strip():
                break
        s = c
        while True:
            c = self.read()
            if c.strip():
                s += c
            else:
                break
        return s

def main():
    input = sys.stdin
    output = sys.stdout
    solver = Solver(input, output)
    solver.solve()

if __name__ == "__main__":
    main()
# End of Code
